hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/HtsCallAddressChecks.java

```diff
@@ -37,7 +37,7 @@ public HtsCallAddressChecks() {
      * @return true if the frame's parent is a delegate call
      */
     public boolean hasParentDelegateCall(@NonNull final MessageFrame frame) {
-        return isDelegateCall(parentOf(frame));
+        return frame.getMessageFrameStack().size() > 1 && isDelegateCall(parentOf(frame));
```

Comment:
This is fine.  Just so you know, I might have put the test over in `parentOf`, having that return an `Optional<MessageFrame`.  Or maybe have a `hasParent()` predicate.

Just a difference in coding style ....